llvm/lib/Target/RISCV/RISCVInstrInfoP.td

```diff
@@ -0,0 +1,1050 @@
+def MQACC_H11  : RVPBinary1LongFW<0b1111, 0b00, 0b111, "mqacc.h11">;
+def MQRACC_H11 : RVPBinary1LongFW<0b1111, 0b10, 0b111, "mqracc.h11">;
+} // Predicates = [HasStdExtP, IsRV32]
+
+let Predicates = [HasStdExtP, IsRV64] in {
+def PMULH_W      : RVPBinary1LongFW<0b0000, 0b01, 0b111, "pmulh.w">;
+def PMULHR_W     : RVPBinary1LongFW<0b0000, 0b11, 0b111, "pmulhr.w">;
+def PMHACC_W     : RVPBinary1LongFW<0b0001, 0b01, 0b111, "pmhacc.w">;
+def PMHRACC_W    : RVPBinary1LongFW<0b0001, 0b11, 0b111, "pmhracc.w">;
+def PMULHU_W     : RVPBinary1LongFW<0b0010, 0b01, 0b111, "pmulhu.w">;
+def PMULHRU_W    : RVPBinary1LongFW<0b0010, 0b11, 0b111, "pmulhru.w">;
+def PMHACCU_W    : RVPBinary1LongFW<0b0011, 0b01, 0b111, "pmhaccu.w">;
+def PMHRACCU_W   : RVPBinary1LongFW<0b0011, 0b11, 0b111, "pmhraccu.w">;
+
+def PMULH_W_H0   : RVPBinary1LongFW<0b0100, 0b01, 0b111, "pmulh.w.h0">;
+def PMULHSU_W_H0 : RVPBinary1LongFW<0b0100, 0b11, 0b111, "pmulhsu.w.h0">;
+def PMHACC_W_H0  : RVPBinary1LongFW<0b0101, 0b01, 0b111, "pmhacc.w.h0">;
+def PMHACCSU_W_H0: RVPBinary1LongFW<0b0101, 0b11, 0b111, "pmhaccsu.w.h0">;
+def PMULH_W_H1   : RVPBinary1LongFW<0b0110, 0b01, 0b111, "pmulh.w.h1">;
+def PMULHSU_W_H1 : RVPBinary1LongFW<0b0110, 0b11, 0b111, "pmulhsu.w.h1">;
+def PMHACC_W_H1  : RVPBinary1LongFW<0b0111, 0b01, 0b111, "pmhacc.w.h1">;
+def PMHACCSU_W_H1: RVPBinary1LongFW<0b0111, 0b11, 0b111, "pmhaccsu.w.h1">;
+
+def PMULHSU_W    : RVPBinary1LongFW<0b1000, 0b01, 0b111, "pmulhsu.w">;
+def PMULHRSU_W   : RVPBinary1LongFW<0b1000, 0b11, 0b111, "pmulhrsu.w">;
+def PMHACCSU_W   : RVPBinary1LongFW<0b1001, 0b01, 0b111, "pmhaccsu.w">;
+def PMHRACCSU_W  : RVPBinary1LongFW<0b1001, 0b11, 0b111, "pmhraccsu.w">;
+def PMULQ_W      : RVPBinary1LongFW<0b1010, 0b01, 0b111, "pmulq.w">;
+def PMULQR_W     : RVPBinary1LongFW<0b1010, 0b11, 0b111, "pmulqr.w">;
+
+def PMQACC_W_H00 : RVPBinary1LongFW<0b1101, 0b00, 0b111, "pmqacc.w.h00">;
+def MQACC_W00    : RVPBinary1LongFW<0b1101, 0b01, 0b111, "mqacc.w00">;
+def PMQRACC_W_H00: RVPBinary1LongFW<0b1101, 0b10, 0b111, "pmqracc.w.h00">;
+def MQRACC_W00   : RVPBinary1LongFW<0b1101, 0b11, 0b111, "mqracc.w00">;
+def PMQACC_W_H11 : RVPBinary1LongFW<0b1111, 0b00, 0b111, "pmqacc.w.h11">;
+def MQACC_W11    : RVPBinary1LongFW<0b1111, 0b01, 0b111, "mqacc.w11">;
+def PMQRACC_W_H11: RVPBinary1LongFW<0b1111, 0b10, 0b111, "pmqracc.w.h11">;
+def MQRACC_W11   : RVPBinary1LongFW<0b1111, 0b11, 0b111, "mqracc.w11">;
+} // Predicates = [HasStdExtP, IsRV64]
+
+
+let Predicates = [HasStdExtP, IsRV32] in {
+def PWSLLI_B : RVPUnary0F0Rdp<0b000, 0b0010000, "pwslli.b">;
+def PWSLLI_H : RVPUnary0F0Rdp<0b001, 0b0100000, "pwslli.h">;
+def WSLLI    : RVPUnary0F0Rdp<0b010, 0b1000000, "wslli">;
+
+def PWSLAI_B : RVPUnary0F0Rdp<0b100, 0b0010000, "pwslai.b">;
+def PWSLAI_H : RVPUnary0F0Rdp<0b100, 0b0100000, "pwslai.h">;
+def WSLAI    : RVPUnary0F0Rdp<0b100, 0b1000000, "wslai">;
+
+def PLI_DH   : RVPUnaryImm9<0b0011000,  "pli.dh", RVPGPRPairRV32>;
```

Comment:
RVPGPRPairRV32 shifts the register before it encodes it but RVPUnaryImm9 uses RVInstIBase which has a 5 bit rd field in bits 11:7. According to the encoding spec this instruction uses bits 11:8 for rd. As this is written, I believe it will put the register>>1 into 10:7.